moto/dynamodb2/models.py

```diff
@@ -110,27 +111,27 @@ def describe_attrs(self, attributes):
         }
 
     def update(self, update_expression, expression_attribute_names, expression_attribute_values):
-        ACTION_VALUES = ['SET', 'set', 'REMOVE', 'remove']
-
-        action = None
-        for value in update_expression.split():
-            if value in ACTION_VALUES:
-                # An action
-                action = value
-                continue
-            else:
+        parts = [p for p in re.split(r'\b(SET|REMOVE|ADD|DELETE)\b', update_expression) if p]
+        for action, valstr in zip(parts[:-1:2], parts[1::2]):
```

Comment:
While true that `zip` behaves that way, we are zipping a list against itself here, so the lengths can't be radically different. At worst, the length is odd and we'll miss the final element. I added an assert to catch that.